java/com/engflow/notificationqueue/Client.java

```diff
@@ -0,0 +1,207 @@
+                finishLatch.countDown();
+              }
+            });
+
+    try {
+      requestObserver.onNext(
+          PullNotificationRequest.newBuilder()
+              .setQueue(QueueId.newBuilder().setName(queueName).build())
+              .build());
+    } catch (RuntimeException e) {
+      // Cancel RPC
+      requestObserver.onError(e);
+      throw e;
+    }
+
+    finishLatch.await();
+  }
+
+  public static void getInvocations(ManagedChannel channel, String invocationId)
+      throws InterruptedException {
+    EventStoreGrpc.EventStoreStub asyncStub = EventStoreGrpc.newStub(channel);
+    asyncStub.getInvocation(
+        GetInvocationRequest.newBuilder().setInvocationId(invocationId).build(),
+        new StreamObserver<StreamedBuildEvent>() {
+          @Override
+          public void onNext(StreamedBuildEvent response) {
+            System.out.println("Invocation: " + response.toString());
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            System.err.println("Error on request: " + t.getMessage());
+          }
+
+          @Override
+          public void onCompleted() {
+            System.out.println("Finished pulling invocation");
+          }
+        });
+  }
+
+  private static ManagedChannel createChannel(
+      String endpoint,
+      String tlsTrustedCertificate,
+      @Nullable String clientCertificate,
+      @Nullable String clientKey) {
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(endpoint),
+        "Empty --notification_queue_endpoint, expected \"protocol://host[:port]\"");
+
+    boolean tls = endpoint.startsWith("grpcs://");
+    Preconditions.checkArgument(
+        tls || endpoint.startsWith("grpc://"),
+        "Bad --notification_queue_endpoint value \""
+            + endpoint
+            + "\", expected \"grpc://\" or \"grpcs://\" protocol");
+    endpoint = endpoint.substring(tls ? "grpcs://".length() : "grpc://".length());
+
+    NettyChannelBuilder builder =
+        NettyChannelBuilder.forTarget(endpoint)
+            .negotiationType(tls ? NegotiationType.TLS : NegotiationType.PLAINTEXT);
+
+    if (tls) {
+      try {
+        SslContextBuilder contextBuilder = GrpcSslContexts.forClient();
+        if (!Strings.isNullOrEmpty(tlsTrustedCertificate)) {
+          File cert = new File(tlsTrustedCertificate);
+          if (!cert.isFile()) {
+            throw new IllegalArgumentException(
+                "\"--tls_trusted_certificate\" must point to an existing file");
+          }
+          contextBuilder = contextBuilder.trustManager(cert);
+        }
+        if (!Strings.isNullOrEmpty(clientCertificate) && !Strings.isNullOrEmpty(clientKey)) {
+          contextBuilder =
+              contextBuilder.keyManager(new File(clientCertificate), new File(clientKey));
+        }
+        builder.sslContext(contextBuilder.build());
+      } catch (SSLException e) {
+        throw new IllegalStateException(e);
```

Comment:
I think `createChannel` should propagate `IOException` and `main` should catch it, print a message, and return. There are a lot of things that can go wrong here that are not under the program's control. That shouldn't crash the program.